### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by JavaEria**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -19,5 +19,4 @@
 npm-debug.log*
 yarn-debug.log*
 yarn-error.log*
-/package.json
 /package-lock.json
```
Comment (ghost) [inline .gitignore]: package lock needs to be committed into the repo. please look into why that should be

File: src/App.js [hunk 2]
```diff
@@ -1,18 +1,44 @@
...
       <div className="App">
         <header className="App-header">
           <img src={logo} className="App-logo" alt="logo" />
-          <h1 className="App-title">Welcome to React</h1>
+          <h1 className="App-title">Todo App</h1>
         </header>
-        <p className="App-intro">
-          To get started, edit <code>src/App.js</code> and save to reload.
-        </p>
+        <InputItem itemCallback = {this.recieveItem}/>
```
Comment (ghost) [inline src/App.js]: itemCallback isnt a very good name. We should name them according to their action. In this case InputItem is either changing or submitting so onChange or onSubmit. An even better way would be to have a InputTodo which has the action to "add"

File: src/InputItem.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import React, { Component } from 'react';
+
+class InputItem extends Component {
+   
+    createItem = (itemDescription) => {
+       return {id: Math.random().toString(36).substr(2, 9), name: itemDescription, isChecked: false};
```
Comment (ghost) [inline src/InputItem.js]: would recommend using some js package for this like uuid

File: src/InputItem.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    }
+    
+
+    sendItem = (event) => {
+        event.preventDefault();
+        this.props.itemCallback(this.createItem(event.target[0].value));
```
Comment (ghost) [inline src/InputItem.js]: target[0] is not good practice, please use target.elements.inputItem to be specific

File: src/TodoItem.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import React, {Component} from 'react';
+import { Checkbox, Button } from 'react-bootstrap'
+
+
+class TodoItem extends React.Component {
+    render() {
+        return (
+            <div>
+               <Checkbox style={{padding: 5, border : '1px solid black', width: 300, margin: '0 auto'}} type="checkbox" checked ={this.props.todo.isChecked ? "checked" : ""} onChange ={() => this.props.selectedCallback(this.props.todo)}>
```
Comment (ghost) [inline src/TodoItem.js]: avoid inline styling

File: src/TodoItem.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+               <p style={{padding: 20, display: 'inline'}}>{this.props.todo.name}</p>
+               <Button onClick = {() => this.props.removeCallback(this.props.todo)}>X</Button>
```
Comment (ghost) [inline src/TodoItem.js]: you can just have onClick = { this.props.removeCallback }

File: src/TodoList.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+    itemSelected = (item) => {
+        this.props.selectTodoCallback(item);
+    }
+
+    render(){
+        return (
+        <div>
+            {this.props.todos.map((item) => (
+                <TodoItem key = {item.id} todo = {item} removeCallback = {this.itemDeleted} selectedCallback = {this.itemSelected}/>
```
Comment (ghost) [inline src/TodoList.js]: consider renaming your props

**Comment by ghost** [comment]
For the next stage lets focus on:

- [ ] Edit todo
- [ ] Complete todo
- [ ] Filter todo (all, incomplete, complete)

File: src/TodoItem.js (hunk 6 above)
Comment (JavaEria) [reply src/TodoItem.js]: We need to pass an argument hence we used the arrow function, it doesn't work without it. We used this reference.
https://reactjs.org/docs/faq-functions.html#how-do-i-pass-a-parameter-to-an-event-handler-or-callback

**Pull request merged**